_official-jokes/app/routes/jokes.tsx

```diff
@@ -7,12 +7,7 @@ import { getUser } from "~/utils/session.server";
 
 import stylesUrl from "../styles/jokes.css";
 
-type LoaderData = {
-  user: Awaited<ReturnType<typeof getUser>>;
-  jokeListItems: Array<{ id: string; name: string }>;
-};
-
-export const loader: LoaderFunction = async ({ request }) => {
+export const loader = async ({ request }: LoaderArgs) => {
```

Comment:
I'd suggest switching to `DataFunctionArgs` which can be used for both loaders and actions. Also, I'd personally switch this to a function declaration.

Reply:
> I'd suggest switching to DataFunctionArgs which can be used for both loaders and actions.

I rather prefer to use `action`/`loader` specific types tbh as that makes it more explicit imo & it's also more in line with `ActionFunction`/`LoaderFunction`
But that's a personal taste

> I'd personally switch this to a function declaration

I opted to keep `action`s/`loader`s the way they are, to make the diff as small as possible.
Also: as you can see in the discussions of https://github.com/remix-run/remix/pull/4675, `satifies` only works with arrow function

Reply:
If `satisfies` only works for arrow functions then I'll probably not use it for this myself 🙁

In any case, I'm not really a maintainer anymore so you all can feel free to do whatever you'd like.

Reply:
Let's wait and see what @mcansh & the rest of the team decide

Reply:
> I'd suggest switching to DataFunctionArgs which can be used for both loaders and actions. Also, I'd personally switch this to a function declaration.

that's kind of where i'm at currently in random side project apps

> If satisfies only works for arrow functions then I'll probably not use it for this myself 🙁

it's only for variables/arrow functions unfortunately 😢, but we're also still on esbuild 0.15.11 due to some stdout issues in tests, so we don't support `satisfies` yet anyways